File: #5397 키로거/김서로.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class 김서로 {

	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
		
		int T = Integer.parseInt(input.readLine());
		
		for (int i=0; i<T; i++) {
			String password = input.readLine();
			Stack<Character> leftStack = new Stack<>();
            Stack<Character> rightStack = new Stack<>();
            
            for (char c: password.toCharArray()) {
            	if (Character.isLetterOrDigit(c)) {
            		leftStack.add(c);
            	} else if (!(leftStack.isEmpty()) && c == '<') {
            		rightStack.add(leftStack.pop());
            	} else if (!(rightStack.isEmpty()) && c == '>') {
            		leftStack.add(rightStack.pop());
            	} else if (!(leftStack.isEmpty()) && c == '-') {
            		leftStack.pop();
            	}
            }
            
            while (!(rightStack.isEmpty())) {
            	leftStack.add(rightStack.pop());
            }
            
            StringBuilder output = new StringBuilder();
            
            for (char c: leftStack) {
            	output.append(c);
            }
            
            System.out.println(output);
		}
	}
}

File: wk01-stack-queue/#5397 키로거/김서로.py
T = int(input())

for _ in range(T):
    inputs = list(input())
    inputs.reverse()
    left_stack = []
    right_stack = []
    while inputs:
        c = inputs.pop()
        if c.isalnum():
            left_stack.append(c)
        if left_stack and c == '<':
            right_stack.append(left_stack.pop())
        if right_stack and c == '>':
            left_stack.append(right_stack.pop())
        if left_stack and c == '-':
            left_stack.pop()
    while right_stack:
        left_stack.append(right_stack.pop())
    print("".join(left_stack))
